### DISCUSSION TITLE
Tag, BaseTag and Commits properties are not being filled in branches other than master

### DISCUSSION THREAD

**Discussion by luucasgb** [devlooped/GitInfo#277] [Q&A]
In my master branch those properties are populated normally. They're not correctly populated in a different branch however.
Latest commit in master is tag 1.5.20. After creating a branch based on 1.5.20 tag for 1.5.21 development, these properties won't update, either being left empty in the case of Tag/BaseTag or being defaulted to 0 in Commits' case.

Any idea of what could be happening? I have the sense that is not a problem with GitInfo but really a problem with my understanding of how git tagging works. Would appreciate any kind of input. Thanks!

**Comment by lucas-0liveira** [comment]
It sounds like there might be an issue with the branch you created for 1.5.21 development. It's possible that there was a problem during the branching process or some specific configuration that's affecting the properties.

Check Branch Creation: Ensure that the branch for 1.5.21 was created properly and that it's based on the 1.5.20 tag. You might want to double-check the branch creation command or process to make sure it's correct.

Inspect Code Changes: Look at the code changes in the new branch. Specifically, focus on any changes related to the properties you mentioned (Tag/BaseTag and Commits). Make sure that there aren't any accidental modifications causing this behavior.

Review Configuration: Check if there's any specific configuration or environment setup that's different in the new branch compared to the master branch. Sometimes, environment variables or configurations can affect how properties are populated.

Verify Data Sources: Ensure that the data sources for these properties are accessible and providing the correct information in the 1.5.21 branch. It's possible that there's an issue with the data retrieval process.

Debugging: Use debugging tools or logs to trace the flow of data and see where it might be getting stuck or overwritten.

Version Control History: Go through the commit history of the new branch and see if there are any specific changes that might have caused this behavior. Look for commits that involve the properties in question.

Compare Configurations: Compare the configurations or settings between the master branch and the new branch to identify any discrepancies that might be causing this issue.

Consult Team Members: If you're working in a team, it's a good idea to discuss this with your colleagues. They might have encountered a similar issue or have insights into the specific codebase.

**Comment by kzu** [comment]
@luucasgb the best way to understand how it works, is to run a build with `-bl`, use msbuildlog.com viewer, and inspect all `Exec` tasks, which are the shell commands run by gitinfo to determine the values. You can run the same commands yourself in your repo root and see why they are/aren't what you expect.
